src/imageio/format/j2k.c

```diff
@@ -574,29 +560,41 @@ void *legacy_params(dt_imageio_module_format_t *self,
   }
 
   // incremental update supported:
-  /*
   if(old_version = 2)
   {
     // let's update from 2 to 3
     typedef struct dt_imageio_j2k_v3_t
```

Comment:
That's correct, but for future update from v3 to v4 it will be needed to use `dt_imageio_j2k_v3_t` as old version, so this decl needs to be just before `if(old_version == 2)`.

Reply:
I see that you have fixed the `=` vs `==,` I was commenting about this but you changed it :) I have fixed all others wrong commented out code.

Reply:
I'll merge and do the clean-up. Thanks.